pkg/controller/directvolumemigration/rsync.go

```diff
@@ -131,16 +133,28 @@ func (t *Task) generateSSHKeys() error {
 	return nil
 }
 
-func (t *Task) createRsyncConfig() error {
+func (t *Task) createRsyncConfig() (bool, error) {
```

Comment:
Why the extra boolean return?

Reply:
It looks like if `err` is `nil` the boolean is always true, and it's always `false` if err is not nil. So it feels redundant

Reply:
@dymurray  no it's not, we are returning `false` and no error when the password is not found. Extra boolean confirm whether the secret with the password is created or not. @alaypatel07 suggested that we first look for a secret if the secret is not found create one and requeue, if the secret is found read the password from it and execute the rest of the logic. This was required because creating a secret and reading from the same secret needed some delay in between.

Reply:
I see.... so `createRsyncConfig` now returns a bool that determines whether we can safely continue on to the next task. Is that acccurate?

Reply:
Yes.